app/models/email_authorization.rb

```diff
@@ -0,0 +1,11 @@
+class EmailAuthorization < ApplicationRecord
+  belongs_to :user
+
+  TYPES = %w[merge_request account_lockout uuid_issue to_be_determined].freeze
```

Comment:
Basically we're going to ask the user to authorize us to do those operations? `to_be_determined` seems like to not be useful though.

a curiosity, what is `uuid_issue` ?

Reply:
Also `TYPES` and `type_of` could be just `kind` which is a good synonym for `type` as we can't use that word

Reply:
> `to_be_determined`
Ah oops, `to_be_determined` was a placeholder hehe

> `uuid_issue`
The `uuid_issue` happens when:
1. a user makes a DEV account with Twitter/GitHub with email `andy@email.com`
2. they delete their Twitter/GitHub account for whatever reason
3. they make a new Twitter/GitHub account with the same email `andy@email.com`
4. they try to log into DEV but fail because their auth data UUID doesn't match the one we have

> type_of
Yeah, I like `kind` too but we have a convention now in the database so IMO better to stick with `type_of`. I think a good task for a different PR is to add the `kind` alias to all models with `type_of`?

Reply:
Let's keep `type_of`, you're right! :) Should the explanation of `uuid_issue` you just gave me be somewhere in the code? So it doesn't get lost in the PR and people won't ask you in the future what `uuid_issue` means (and by people I mean me :D)